src/swap/hooks/useAwaitCalls.ts

```diff
@@ -0,0 +1,46 @@
+import { useCallback } from 'react';
+import { waitForTransactionReceipt } from 'wagmi/actions';
+import { useCallsStatus } from 'wagmi/experimental';
+import type { UseAwaitCallsParams } from '../types';
+
+export function useAwaitCalls({
+  accountConfig,
+  config,
+  lifecycleStatus,
+  setLifecycleStatus,
+}: UseAwaitCallsParams) {
+  const callsId =
+    lifecycleStatus.statusName === 'transactionApproved'
+      ? lifecycleStatus.statusData?.callsId
+      : undefined;
+
+  const { data } = useCallsStatus({
+    id: callsId || '',
+    query: {
+      refetchInterval: (query) => {
+        return query.state.data?.status === 'CONFIRMED' ? false : 1000;
+      },
+      enabled: callsId !== undefined,
+    },
+  });
+
+  return useCallback(async () => {
+    if (data?.status === 'CONFIRMED' && data?.receipts) {
+      const transactionReceipt = await waitForTransactionReceipt(
+        accountConfig,
+        {
+          confirmations: 1,
+          hash: data.receipts[data.receipts.length - 1].transactionHash,
```

Comment:
how many receipts would be here?

Reply:
only 1 I believe... I'm not sure which cases have more than 1 receipt

Reply:
but default to returning the last in those cases

Reply:
what is the difference between the receipts contained in data object returned from `useCallsStatus` and the receipt returned from `waitForTransactionReceipt` here?

Reply:
the former is a different object, we're expecting a `TransactionReceipt`

```
export type WalletCallReceipt<quantity = Hex, status = Hex> = {
  logs: {
    address: Hex
    data: Hex
    topics: Hex[]
  }[]
  status: status
  blockHash: Hex
  blockNumber: quantity
  gasUsed: quantity
  transactionHash: Hex
}
```

Reply:
is there any data the user gets from second receipt that they wouldn't be able to get from first receipt ? just asking because i remember a conversation with Wilson where he brought up that it might not be worth the extra latency from second request. sending you the slack link in DM.